### ISSUE TITLE
`pcluster create` fails to add more than two compute nodes

### ISSUE THREAD

**Issue by armaneshaghi** [aws/aws-parallelcluster#987]
**Environment:**
 - AWS ParallelCluster: current dev version.
 - OS: ALinux
 - Scheduler: SGE
 - Master instance type:C4.large
 - Compute instance type: C4.xlarge

**Bug description and how to reproduce:**
Cluster creation fails after adding 2 compute nodes. The cluster, however, seems functional with the two nodes that are created. 

```
$ pcluster create cluster1
Creating stack named: parallelcluster-cluster1
Status: parallelcluster-cluster1 - ROLLBACK_IN_PROGRESS                         
Cluster creation failed.  Failed events:
  - AWS::AutoScaling::AutoScalingGroup ComputeFleet Received 2 SUCCESS signal(s) out of 20.  Unable to satisfy 100% MinSuccessfulInstancesPercent requirement

```
**Additional context:**
Config file

```
[global]                                                                                                                                               
cluster_template = default
update_check = true
sanity_check = true

[aws]
aws_region_name = eu-west-2
aws_access_key_id = <REMOVED>
aws_secret_access_key = <REMOVED>

[cluster default]
vpc_settings = public
key_name = key
master_instance_type = c4.large
compute_instance_type = c4.xlarge
cluster_type = spot
spot_price = 0.2 
master_root_volume_size = 17
initial_queue_size = 20
max_queue_size = 1000

custom_ami = ami-<REMOVED>
shared_dir = /software
ebs_settings = custom1
scaling_settings = custom

[vpc public]
vpc_id = <REMOVED>
master_subnet_id = subnet-<REMOVED>
ssh_from = <REMOVED>/32

[aliases]
ssh = ssh {CFN_USER}@{MASTER_IP} -i ~/key.pem
vpc_settings = public

[scaling custom]
scaledown_idletime = 10

[ebs custom1]
shared_dir = /software
volume_size = 60
ebs_volume_id = vol-<REMOVED>                        
```

 - pre/post-install scripts: None
- I have attached `/var/log/cfn-init.log`, `/var/log/cloud-init-output.log`, and `/var/log/cloud-init.log` from Master node 
 - `/var/log/jobwatcher` and `/var/log/sqswatcher` from the Master node are attached as well. 
[cfn-init.log](https://github.com/aws/aws-parallelcluster/files/3045777/cfn-init.log)
[cloud-init-output.log](https://github.com/aws/aws-parallelcluster/files/3045779/cloud-init-output.log)
[cloud-init.log](https://github.com/aws/aws-parallelcluster/files/3045782/cloud-init.log)

[jobwatcher.txt](https://github.com/aws/aws-parallelcluster/files/3045783/jobwatcher.txt)
[sqswatcher.txt](https://github.com/aws/aws-parallelcluster/files/3045784/sqswatcher.txt)

Below `/var/log/messages` from master node in case it is useful (added .txt so that Github allows upload).
[messages.txt](https://github.com/aws/aws-parallelcluster/files/3045804/messages.txt)

```
Master node$ qhost
HOSTNAME                ARCH         NCPU NSOC NCOR NTHR  LOAD  MEMTOT  MEMUSE  SWAPTO  SWAPUS
----------------------------------------------------------------------------------------------
global                  -               -    -    -    -     -       -       -       -       -
ip-172-31-12-147        lx-amd64        4    1    2    4  0.00    7.3G  156.7M     0.0     0.0
ip-172-31-9-254         lx-amd64        4    1    2    4  0.02    7.3G  154.4M     0.0     0.0
```

`nodewatcher` from one of the compute nodes. 
```
2019-04-04 23:01:07,624 INFO [nodewatcher:main] nodewatcher startup
2019-04-04 23:01:07,624 INFO [nodewatcher:_get_config] Reading /etc/nodewatcher.cfg
2019-04-04 23:01:07,625 INFO [nodewatcher:_get_config] Configured parameters: region=eu-west-2 scheduler=sge stack_name=parallelcluster-cluster1 scaledown_idletime=10 proxy=NONE
2019-04-04 23:01:07,628 INFO [nodewatcher:main] Instance id is i-09de620821371c0c9, hostname is ip-172-31-12-147.eu-west-2.compute.internal
2019-04-04 23:01:07,829 INFO [utils:get_asg_name] ASG parallelcluster-cluster1-ComputeFleet-1BL1HRJWM5J9Q found for the stack parallelcluster-cluster1
2019-04-04 23:02:07,890 INFO [nodewatcher:_is_stack_ready] Checking for status of the stack parallelcluster-cluster1
2019-04-04 23:02:08,052 INFO [nodewatcher:_poll_instance_status] Stack parallelcluster-cluster1 ready: False
2019-04-04 23:03:08,113 INFO [nodewatcher:_is_stack_ready] Checking for status of the stack parallelcluster-cluster1
2019-04-04 23:03:08,192 INFO [nodewatcher:_poll_instance_status] Stack parallelcluster-cluster1 ready: False
2019-04-04 23:04:08,252 INFO [nodewatcher:_is_stack_ready] Checking for status of the stack parallelcluster-cluster1
2019-04-04 23:04:08,311 INFO [nodewatcher:_poll_instance_status] Stack parallelcluster-cluster1 ready: False
2019-04-04 23:05:08,372 INFO [nodewatcher:_is_stack_ready] Checking for status of the stack parallelcluster-cluster1
2019-04-04 23:05:08,439 INFO [nodewatcher:_poll_instance_status] Stack parallelcluster-cluster1 ready: False
2019-04-04 23:06:08,499 INFO [nodewatcher:_is_stack_ready] Checking for status of the stack parallelcluster-cluster1
2019-04-04 23:06:08,571 INFO [nodewatcher:_poll_instance_status] Stack parallelcluster-cluster1 ready: False
2019-04-04 23:07:08,631 INFO [nodewatcher:_is_stack_ready] Checking for status of the stack parallelcluster-cluster1
2019-04-04 23:07:08,703 INFO [nodewatcher:_poll_instance_status] Stack parallelcluster-cluster1 ready: False
2019-04-04 23:08:08,763 INFO [nodewatcher:_is_stack_ready] Checking for status of the stack parallelcluster-cluster1
2019-04-04 23:08:08,830 INFO [nodewatcher:_poll_instance_status] Stack parallelcluster-cluster1 ready: False
2019-04-04 23:09:08,891 INFO [nodewatcher:_is_stack_ready] Checking for status of the stack parallelcluster-cluster1
2019-04-04 23:09:08,954 INFO [nodewatcher:_poll_instance_status] Stack parallelcluster-cluster1 ready: False
```

**Comment by armaneshaghi** [comment]
I would like to add the issue above is resolved if I change the region to us-east-1. So could it be there is limitation in specific areas (here eu-west-1) or somehow my subnet addresses are restricted?

**Comment by sean-smith** [comment]
Hmm, I think this an issue with the `spot_price` / `initial_queue_size`. When your cluster goes to to bootstrap, it needs to bring up all 20 compute instances, at or below the `spot_price` you specified or cluster bootstrap or it will fail. Due to the nature of the spot market in `eu-west-1`, it probably can't get all 20 at once. For example, it could get 19 and then stack creation (which is controlled through cloudformation) could time out.

My suggestion is to change `initial_queue_size = 20` to something small (~2), then after the cluster creates, add the extra needed nodes to the ASG, or just submit your jobs and it'll autoscale.

To do so, create the cluster, then go to the [EC2 Console](https://console.aws.amazon.com/ec2/autoscaling/) > Auto Scaling Groups > `parallelcluster-[cluster_name]` click Actions > edit and change Desired Capacity to be `20`. 

We recently made a change so that the `spot_price` parameter isn't necessary. If you don't specify a spot price, then it uses the current spot market price capped at on-demand. My theory is that if you remove the `spot_price` it'll have better luck getting all 20 instances before one gets reclaimed. You can read more about those changes here: https://aws.amazon.com/blogs/compute/new-amazon-ec2-spot-pricing/

**tl;dr** set `initial_queue_size = 2` and remove `spot_price = 0.2`

**Comment by armaneshaghi** [comment]
@sean-smith many thanks. In addition to the above thoughts, just to add for posterity there is also a default limit on the number of spot instances a user can initiate. This limit (~20) is assessed against previous recent attempts to bid for a spot instance, meaning that even if the number of requests is less than or equal to the limit but adds to more than the limit in a short time those will not spawn any instances. The solution is to get in touch with AWS support.

**Comment by enrico-usai** [comment]
Hi @armaneshaghi , thank you for sharing.
